go.mod

```diff
@@ -1,12 +1,10 @@
 module go.mau.fi/whatsmeow
 
-go 1.23.0
-
-toolchain go1.24.0
+go 1.24
```

Comment:
Go 1.22 and up require the patch version in the `go` statement, so `go 1.24` is not valid. The `toolchain` statement is also needed so that the auto-update mechanism would get the latest version rather than the oldest supported version.